File: test_opencode_connection.py

import requests

base_url = "http://127.0.0.1:4096"

print(f"Testing connection to: {base_url}")

try:
    # Test 1: Simple GET to baseUrl (like health check)
    print("\n--- Test 1: Simple GET ---")
    response1 = requests.get(base_url, timeout=3)
    print(f"Status: {response1.status_code}")
    
    # Test 2: Test with /api/agent path (like the actual API call)
    print("\n--- Test 2: GET /api/agent ---")
    response2 = requests.get(f"{base_url}/api/agent", timeout=3)
    print(f"Status: {response2.status_code}")
    print(f"Response: {response2.text[:200]}...")
    
    # Test 3: Test with a workspaceId parameter
    print("\n--- Test 3: GET /api/agent?workspaceId=test ---")
    response3 = requests.get(f"{base_url}/api/agent", params={"workspaceId": "test"}, timeout=3)
    print(f"Status: {response3.status_code}")
    print(f"Response: {response3.text[:200]}...")
    
except requests.exceptions.RequestException as e:
    print(f"\nError: {type(e).__name__}: {e}")
    import traceback
    traceback.print_exc()

File: TestOpencodeConnection.java

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.time.Duration;

public class TestOpencodeConnection {
    public static void main(String[] args) {
        String baseUrl = "http://127.0.0.1:4096";
        System.out.println("Testing connection to: " + baseUrl);
        
        HttpClient httpClient = HttpClient.newBuilder()
                .connectTimeout(Duration.ofSeconds(2))
                .build();
        
        try {
            // Test 1: Simple GET to baseUrl (like health check)
            System.out.println("\n--- Test 1: Simple GET ---");
            HttpRequest request1 = HttpRequest.newBuilder()
                    .uri(URI.create(baseUrl))
                    .timeout(Duration.ofSeconds(3))
                    .GET()
                    .build();
            HttpResponse&lt;Void&gt; response1 = httpClient.send(request1, HttpResponse.BodyHandlers.discarding());
            System.out.println("Status: " + response1.statusCode());
            
            // Test 2: Test with /api/agent path (like the actual API call)
            System.out.println("\n--- Test 2: GET /api/agent ---");
            HttpRequest request2 = HttpRequest.newBuilder()
                    .uri(URI.create(baseUrl + "/api/agent"))
                    .timeout(Duration.ofSeconds(3))
                    .GET()
                    .build();
            HttpResponse&lt;String&gt; response2 = httpClient.send(request2, HttpResponse.BodyHandlers.ofString());
            System.out.println("Status: " + response2.statusCode());
            System.out.println("Response: " + response2.body().substring(0, Math.min(200, response2.body().length())) + "...");
            
        } catch (Exception e) {
            System.err.println("Error: " + e.getClass().getName() + ": " + e.getMessage());
            e.printStackTrace();
        }
    }
}
